cmake/Scripts/GenerateSleefHeader.cmake

```diff
@@ -19,23 +10,31 @@ if(OPTION_SHOW_CONFIG)
 endif(OPTION_SHOW_CONFIG)
 
 # Generate header for x86
+  set(run1 2 4 __m128d __m128 __m128i __m128i __SSE2__)
+  set(run2 2 4 __m128d __m128 __m128i __m128i __SSE2__ sse2)
+  set(run3 2 4 __m128d __m128 __m128i __m128i __SSE2__ sse4)
+  set(run4 2 4 __m128d __m128 __m128i __m128i __SSE2__ avx2128)
+  set(run5 4 8 __m256d __m256 __m128i "struct { __m128i x, y\; }" __AVX__)
+  set(run6 4 8 __m256d __m256 __m128i "struct { __m128i x, y\; }" __AVX__ avx)
+  set(run7 4 8 __m256d __m256 __m128i "struct { __m128i x, y\; }" __AVX__ fma4)
+  set(run8 4 8 __m256d __m256 __m128i __m256i __AVX__ avx2)
+  set(run9 8 16 __m512d __m512 __m256i __m512i __AVX512F__ avx512f)
+
 # TODO: Change condition of generation to use a COMPILER_SUPPORTS_* flag
 if(SLEEF_ARCH_X86)
-  list(APPEND MKRENAME_PARAMS_LIST
-    "2 4 __m128d __m128 __m128i __m128i __SSE2__")
-  list(APPEND MKRENAME_PARAMS_LIST
-    "2 4 __m128d __m128 __m128i __m128i __SSE2__ sse2")
-  list(APPEND MKRENAME_PARAMS_LIST
-    "2 4 __m128d __m128 __m128i __m128i __SSE2__ sse4")
+  list(APPEND PARAMS_POINTER_LIST run1 run2 run3 run4 run5 run6 run7 run8 run9)
 endif()
 
-foreach(params_set ${MKRENAME_PARAMS_LIST})
-  run_mkrename(${params_set} RESULT_VARIABLE)
-  set(OUTPUT "${OUTPUT}\n${RESULT_VARIABLE}")
+foreach(params_set ${PARAMS_POINTER_LIST})
```

Comment:
I am sure I am missing something, but how comes that the param_set is grouping the params correctly? Could you explain this in the comments?